### ISSUE TITLE
E-Commerce framework pick

### ISSUE THREAD

**Issue by WebDevJL** [Puzzlout/TrekToursStorefront#3]
@damirius This is my answer to your question about the subject:

> We haven't talked too much about storefront architecture code wise. In any case, I assume that we want to keep API/Client relation. So we want to use that API with new client app that will be storefront right? If that's the case where exactly do we want to use some existing e-commerce solution? For the client or for the API?

I'd like to keep the API/Client relation yes. For now I know something for sure: we either build from nothing a e-commerce solution or we use something done and tweak if necessary. I can't answer your other questions as per what the e-commerce solutions I quoted to you do. That's why I'd like you to look at it. Maybe there is a API/Client separation already in one of them, maybe not.
If, while we investigate the e-commerce solution, we see that none is mature enough but could be a starting point, then we will either improve the best one. Or build something from scratch in the worst case scenario.

The choice here is the same as the one I made when we started the flyer: I had my "infant" framework and I had Symphony. I chose Symphony because it was mature. I'd hope we can use some existing base because e-commerce applications are complex. If you know another one that have Symphony as base, let me know.

**Comment by damirius** [comment]
@WebDevJL 

This is rather big choice we have to make. Mostly it depends on what we want to accomplish. If that documents regarding storefront we have in gdrive contain everything we need to know about it, I'll go and reread them to see what exactly do we want to accomplish. Since we already have working API, we should integrate everything seamlessly with it. If there's a lot of stuff we need to create for it we can go with some existing ecommerce. Sylius is definitely my favorite for now since it can be split in bundles: http://docs.sylius.org/en/latest/bundles/index.html. We can use whatever we wish and create our custom controllers in the API.

**Comment by WebDevJL** [comment]
@damirius 

No, the document is only a very succinct draft of the functionalities. I actually need to work more on it.
It tries to define a product to the smallest entity.

I think that the e-commerce part will be just standard e-commerce stuff (user account, basket, payment, product pages, etc), hence using something built instead of building it from scratch. 

The other very important task at hand is **_content management of the products**_ (i.e. organized trekking tours). In fact, that will be the administrator(s) task and the admin area will grow a lot! 
External tours organizers will need to be allows to come setup their products on the long term.

So maybe, the e-commerce isn't the only thing to investigate and we may just stylius (which I also prefer so far) but also look into [Symphony CMF](http://cmf.symfony.com/) ? What do you think?

**Comment by WebDevJL** [comment]
@damirius 

I have two documents to share with you to start investigation about the technical solution to use in order to build the storefront:
1. A preliminary data model of the entities we will need: #4 
2. A preliminary list of controllers and their actions: #5 

I think we will deduce the endpoints between the storefront app and its API once we have the data model and controller/action listings done.

Let me know what you think.

**Comment by WebDevJL** [comment]
Available options:
- Sylius : http://sylius.org/
- Thelia : e-commerce based on Symphony (http://www.thelia.fr/)
- Elcodi : http://elcodi.io/

Review : https://www.symfony.fi/entry/symfony-and-ecommerce-elcodi-sylius-and-thelia

**Comment by damirius** [comment]
@WebDevJL 
As discussed we will probably integrate Sylius bundles when needed http://docs.sylius.org/en/latest/bundles/index.html
We will review them in details too see exactly what we will get and how much they will represent what we want

**Comment by WebDevJL** [comment]
@damirius That's good. We will look at this once we better know what we
need (see #4 and #5).

## 

Jérémie Litzer
CEO @ _Puzzlout_ http://puzzlout.com
